This is looking good but could these new failures due to tests running too slow have been caused by the added overhead of getting the `sample_shape` from the `full_shape`? 

```
 ================================== FAILURES ===================================
_____ test_sample_against_cdf[Halton-sample-dist_shape0-x_shape1-Normal] ______
[gw1] win32 -- Python 3.12.9 C:\hostedtoolcache\windows\Python\3.12.9\x64\python.exe
Test passed but took too long to run: Duration 2.1438161000005493s > 1.0s
______ test_sample_against_cdf[Sobol-sample-dist_shape0-x_shape1-Normal] ______
[gw1] win32 -- Python 3.12.9 C:\hostedtoolcache\windows\Python\3.12.9\x64\python.exe
Test passed but took too long to run: Duration 1.9654189000002589s > 1.0s
```

I did a rough benchmark, and saw the following on `main`

```
In [1]: import scipy.stats as stats

In [2]: import scipy.stats.qmc as qmc

In [3]: rng = qmc.Halton(d=1, seed=1234)

In [4]: dist = stats.Normal()

In [5]: %timeit dist.sample((100, 100), rng=rng)
10.4 ms ± 3.01 μs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```

and this on this PR branch:

```
In [1]: import scipy.stats as stats

In [2]: import scipy.stats.qmc as qmc

In [3]: rng = qmc.Halton(d=1, seed=1234)

In [4]: dist = stats.Normal()

In [5]: %timeit dist.sample((100, 100), rng=rng)
883 ms ± 1.29 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

Were you expecting such a slowdown?

--- a/scipy/stats/_distribution_infrastructure.py
+++ b/scipy/stats/_distribution_infrastructure.py
@@ -2884,7 +2884,7 @@ class ContinuousDistribution(_ProbabilityDistribution):
         # Generate QMC uniform sample(s) on unit interval with specified shape;
         # if `sample_shape != ()`, then each slice along axis 0 is independent.
 
-        sample_shape = full_shape[:-len(self._shape)]
+        sample_shape = full_shape[:len(full_shape)-len(self._shape)]
         # Determine the number of independent sequences and the length of each.
         n_low_discrepancy = sample_shape[0] if sample_shape else 1
         n_independent = math.prod(full_shape[1:] if sample_shape else full_shape)

